libsolidity/formal/SMTEncoder.cpp

```diff
@@ -2826,18 +2844,18 @@ void SMTEncoder::createReturnedExpressions(FunctionCall const& _funCall)
 		defineExpr(_funCall, currentValue(*returnParams.front()));
 }
 
-vector<smtutil::Expression> SMTEncoder::symbolicArguments(FunctionCall const& _funCall)
+vector<smtutil::Expression> SMTEncoder::symbolicArguments(FunctionCall const& _funCall, ContractDefinition const* _contract)
```

Comment:
@blishko We actually need the FunctionDefinition here to know exactly how many and what types it expects. If we only use the FunctionType from the FunctionCall, it gives the wrong numbers if the function is bound.

Reply:
Yes, I agree. It would only be possible if `SMTEncoder::expr` would not require `targetType` for proper encoding...